### PR TITLE
request param

### PR DESCRIPTION
**Pull request by orthagonal**

### CODE REVIEW

File: methods/addReport.js [hunk 1]
```diff
@@ -8,6 +8,11 @@ module.exports = {
     if (fn.options && typeof fn.options.cache === 'function') {
       fn.options = fn.options || {};
       fn.options.cache = fn.options.cache(this, this.settings.app);
+      // hapi's default generateKey method can't handle the request object parameter
+      // so if one is not provided it must be overridden
+      if (!fn.options.generateKey) {
+        fn.options.generateKey = (request, ...args) => args.join(',');
```
Comment (jgallen23) [inline methods/addReport.js]: the cache key should just be the report name and stringified query

**Review by jgallen23 — CHANGES_REQUESTED** [review]

File: package.json [hunk 2]
```diff
@@ -4,7 +4,7 @@
   "description": "library to help create a reports service",
   "main": "cli.js",
   "scripts": {
-    "test": "eslint . && tap --cov test"
+    "test": "NODE_ENV=test eslint . && tap --cov test/addReport.test.js"
```
Comment (jgallen23) [inline package.json]: ?

**Review by jgallen23 — CHANGES_REQUESTED** [review]

File: package.json [hunk 3]
```diff
@@ -4,7 +4,7 @@
   "description": "library to help create a reports service",
   "main": "cli.js",
   "scripts": {
-    "test": "eslint . && tap --cov test"
+    "test": "eslint . && NODE_ENV=test tap --cov test/addReport.test.js"
```
Comment (jgallen23) [inline package.json]: why just addReport?

**Review by jgallen23 — CHANGES_REQUESTED** [review]

File: package.json (hunk 3 above)
Comment (orthagonal) [reply package.json]: It reports a 'skip' for the files in test/report if I do the whole dir, which was annoying me.  addReport.test.js is the only file with tests in it but can change this to *.test.js if you prefer.

Comment (jgallen23) [reply package.json]: yeah, lets do *.test.js

**Pull request merged**